### PR TITLE
Restructure the video/video_reader C++ codebase

### LINKED ISSUE #3155: Restructure and clean up the video/video_reader C++ implementations
## 🐛 Bug

The C++ implementations of `video` and `video_reader` need to be refactored and cleaned up similar to #2998, #3051, #3094, #3097, #3135, #3163:
- [x] Rename C++ files, methods & variables according to [recommended naming conventions](https://google.github.io/styleguide/cppguide.html#General_Naming_Rules).
- [x] Move C++ code away from header files and into cpp files.
- [x] Ensure each cpp file has a separate header file with "public" functions.
- [x] Reduce unnecessary header inclusions. 
- [x] Add all internal functions in anonymous namespaces.
- [x] Sync variable names between the cpp files and their header files.

### PR DESCRIPTION
**Pull request by datumbox**
Fixes #3155

### CODE REVIEW

File: torchvision/csrc/io/video/video.cpp [hunk 1]
```diff
@@ -319,3 +320,15 @@ std::tuple<torch::Tensor, double> Video::Next() {
...
 }
+
+static auto registerVideo =
+    torch::class_<Video>("torchvision", "Video")
+        .def(torch::init<std::string, std::string>())
+        .def("get_current_stream", &Video::getCurrentStream)
+        .def("set_current_stream", &Video::setCurrentStream)
+        .def("get_metadata", &Video::getStreamMetadata)
+        .def("seek", &Video::Seek)
+        .def("next", &Video::Next);
```
Comment (datumbox) [inline torchvision/csrc/io/video/video.cpp:331]: Moved from register.cpp

File: torchvision/csrc/io/video_reader/video_reader.cpp [hunk 2]
```diff
@@ -417,95 +418,6 @@ torch::List<torch::Tensor> readVideo(
   return result;
 }
 
-torch::List<torch::Tensor> readVideoFromMemory(
```
Comment (datumbox) [inline torchvision/csrc/io/video_reader/video_reader.cpp:420]: Moving out of the anonymous namespace.

File: torchvision/csrc/io/video_reader/video_reader.cpp [hunk 3]
```diff
@@ -650,20 +562,112 @@ torch::List<torch::Tensor> probeVideo(
   return result;
 }
 
-torch::List<torch::Tensor> probeVideoFromMemory(torch::Tensor input_video) {
+} // namespace
+
+torch::List<torch::Tensor> read_video_from_memory(
```
Comment (datumbox) [inline torchvision/csrc/io/video_reader/video_reader.cpp:567]: "public" methods.

File: torchvision/csrc/io/video_reader/video_reader.cpp [hunk 4]
```diff
@@ -650,20 +562,112 @@ torch::List<torch::Tensor> probeVideo(
...
-
 TORCH_LIBRARY_FRAGMENT(video_reader, m) {
-  m.def("read_video_from_memory", video_reader::readVideoFromMemory);
-  m.def("read_video_from_file", video_reader::readVideoFromFile);
-  m.def("probe_video_from_memory", video_reader::probeVideoFromMemory);
-  m.def("probe_video_from_file", video_reader::probeVideoFromFile);
+  m.def("read_video_from_memory", read_video_from_memory);
+  m.def("read_video_from_file", read_video_from_file);
+  m.def("probe_video_from_memory", probe_video_from_memory);
+  m.def("probe_video_from_file", probe_video_from_file);
```
Comment (datumbox) [inline torchvision/csrc/io/video_reader/video_reader.cpp:669]: Registration of methods within the namespaces at the end of the file.

File: torchvision/csrc/io/video_reader/video_reader.cpp [hunk 5]
```diff
@@ -18,8 +14,13 @@ PyMODINIT_FUNC PyInit_video_reader(void) {
 }
 #endif
 
+using namespace ffmpeg;
+
+namespace vision {
```
Comment (datumbox) [inline torchvision/csrc/io/video_reader/video_reader.cpp:19]: Follow similar pattern as `vision::ops`

File: torchvision/csrc/io/video_reader/video_reader.h [hunk 6]
```diff
@@ -0,0 +1,55 @@
+#pragma once
+
+#include <torch/types.h>
+
+namespace vision {
+namespace video_reader {
+
+torch::List<torch::Tensor> read_video_from_memory(
```
Comment (datumbox) [inline torchvision/csrc/io/video_reader/video_reader.h:8]: public methods in header

**Review by datumbox — COMMENTED** [review]
I provide explanations to some modifications below:

**Review by fmassa — APPROVED** [review]
Looks great to me, thanks!

One thing I was thinking was if in the future we would want to move `decoder` inside `video`, and have both `video_reader` and `video.cpp` within the `video` folder.
Something like
```
io/
    image/
    video/
        decoder/
        video.cpp
        video_reader_legacy.cpp (renamed from video_reader)
        ...
```
This is minor though, but just passed through my mind as something which could improve in clarity about the role of the different video decoders

**Comment by datumbox** [comment]
@fmassa Do you think it makes sense to do this now? The next release brings LOTS of changes on the C++ API, so we might as well do that?

**Pull request merged**